### ISSUE TITLE
[Serializer] Undefined property: Proxies\__CG__\App\Entity\ChildEntity::$lazyObjectState

### ISSUE THREAD

**Issue by alexandre-le-borgne** [symfony/symfony#54393]
### Symfony version(s) affected

7.0.4 & 6.4.4

### Description

When serializing a doctrine proxy of an entity and using the ignored attribute on a property of the entity, I get the error :

 User Notice: Undefined property: Proxies\__CG__\App\Entity\ChildEntity::$lazyObjectState
 in /home/../vendor/symfony/property-access/PropertyAccessor.php on line 408
 
 Removing the property with the #[Ignore] attribute fixes the problem.
 

### How to reproduce

https://github.com/alexandre-le-borgne/bug-symfony-phpdoc-serialization

Diff: https://github.com/alexandre-le-borgne/bug-symfony-phpdoc-serialization/commit/07b80b90b4d8e69c28a4f4702302cd5c3926404f

```
// IndexController.php
#[AsController]
class IndexController {
  public function __construct(private SerializerInterface $serializer, private EntityManagerInterface $doctrine)
  {
  }
    
  #[Route('/')]
  public function __invoke()
    $parent = $this->doctrine->find(ParentEntity::class, 1);
    $model = new ExampleModel();
    // Trying to serialize an object containing a proxy from doctrine
    $model->childEntity = $parent->getChildEntity();
    dd($this->serializer->normalize($model));
  }
}
```
```
#[Entity]
class ParentEntity
{
    #[Id]
    #[Column]
    #[GeneratedValue]
    private int $id;

    #[OneToOne(cascade: ['persist'])]
    private ChildEntity $childEntity;


    public function getId(): int
    {
        return $this->id;
    }

    public function getChildEntity(): ChildEntity
    {
        return $this->childEntity;
    }

    public function setChildEntity(ChildEntity $childEntity): void
    {
        $this->childEntity = $childEntity;
    }
}
```
```
#[Entity]
class ChildEntity
{
    #[Id]
    #[Column]
    #[GeneratedValue]
    private int $id;

    #[Ignore]
    public string $ignored = 'ignored';

    public function getId(): int
    {
        return $this->id;
    }
}
```
```
class ExampleModel
{
    /**
     * @var ChildEntity
     */
    public ChildEntity $childEntity;
}
```
```
class ChildEntity
{
    #[Id]
    #[Column]
    #[GeneratedValue]
    private int $id;

    #[Ignore]
    public string $ignored = 'ignored';

    public function getId(): int
    {
        return $this->id;
    }
}
```

Removing the property "$ignored" with the Ignore attribute in the ChildEntity fixes the problem.

### Possible Solution

Add in config/framework.yaml

```
framework:
    serializer:
        default_context:
            ignored_attributes:
                - '__initializer__' 
                - '__cloner__'
                - '__isInitialized__'
                - 'lazyObjectState'
                - 'lazyObjectInitialized'
                - 'lazyObjectAsInitialized'
 ```

### Additional Context

Seems related to the code in AbstractNormalizer.php

```
        $ignoreUsed = false;
        // ...

        if (!$ignoreUsed && [] === $groups && $allowExtraAttributes) {
            // Backward Compatibility with the code using this method written before the introduction of @Ignore
            return false;
        }
        return $allowedAttributes;
```

![image](https://github.com/symfony/symfony/assets/10196942/18a62d49-7822-4ac0-b922-0dc574d2961c)

**Comment by stof** [comment]
Can you try again with the dev version of Symfony ? https://github.com/symfony/symfony/pull/54194 is not included in the 7.0.4 and 6.4.4 releases as it has been merged after them.

**Comment by alexandre-le-borgne** [comment]
@stof Done.

I  ran the command `composer require symfony/property-access:dev -W`
As expected (https://github.com/symfony/symfony/pull/54194#issuecomment-2017657816), the error is still there

**Comment by xabbuh** [comment]
You also need to update `symfony/var-exporter` and `symfony/serializer` to have the fixes from #54224 and #54485 (will be part of the next patch releases). I have tested your example application (thanks for it, that helped a lot 👍) and it works (again) with the patches from the linked PRs.
